src/modules/helpers/index.ts

```diff
@@ -198,27 +235,26 @@ export class SimpleHelpersModule extends SimpleModuleBase {
    * @param string The template string to parse. Defaults to `''`.
    * @param symbol The symbol to replace with digits. Defaults to `'#'`.
    *
+   * @see faker.string.numeric(): For the replacement method.
+   *
    * @example
    * faker.helpers.replaceSymbolWithNumber() // ''
    * faker.helpers.replaceSymbolWithNumber('#####') // '04812'
    * faker.helpers.replaceSymbolWithNumber('!####') // '27378'
    * faker.helpers.replaceSymbolWithNumber('Your pin is: !####') // '29841'
    *
    * @since 2.0.1
+   *
+   * @deprecated Use `faker.string.numeric()` instead. Example: `string.replace(/#+/g, (m) => faker.string.numeric(m.length));`
```

Comment:
Maybe rename the variable to reduce conflicts with `faker.string`?

```suggestion
   * @deprecated Use `faker.string.numeric()` instead. Example: `value.replace(/#+/g, (m) => faker.string.numeric(m.length));`
```